// Code generated by mockery v1.0.0. DO NOT EDIT.

package modelmock

import (
	context "context"

	model "github.com/piotrkowalczuk/charon/internal/model"
	mock "github.com/stretchr/testify/mock"
)

// UserGroupsProvider is an autogenerated mock type for the UserGroupsProvider type
type UserGroupsProvider struct {
	mock.Mock
}

// DeleteByUserID provides a mock function with given fields: ctx, id
func (_m *UserGroupsProvider) DeleteByUserID(ctx context.Context, id int64) (int64, error) {
	ret := _m.Called(ctx, id)

	var r0 int64
	if rf, ok := ret.Get(0).(func(context.Context, int64) int64); ok {
		r0 = rf(ctx, id)
	} else {
		r0 = ret.Get(0).(int64)
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(context.Context, int64) error); ok {
		r1 = rf(ctx, id)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// Exists provides a mock function with given fields: ctx, userID, groupID
func (_m *UserGroupsProvider) Exists(ctx context.Context, userID int64, groupID int64) (bool, error) {
	ret := _m.Called(ctx, userID, groupID)

	var r0 bool
	if rf, ok := ret.Get(0).(func(context.Context, int64, int64) bool); ok {
		r0 = rf(ctx, userID, groupID)
	} else {
		r0 = ret.Get(0).(bool)
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(context.Context, int64, int64) error); ok {
		r1 = rf(ctx, userID, groupID)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// Find provides a mock function with given fields: ctx, expr
func (_m *UserGroupsProvider) Find(ctx context.Context, expr *model.UserGroupsFindExpr) ([]*model.UserGroupsEntity, error) {
	ret := _m.Called(ctx, expr)

	var r0 []*model.UserGroupsEntity
	if rf, ok := ret.Get(0).(func(context.Context, *model.UserGroupsFindExpr) []*model.UserGroupsEntity); ok {
		r0 = rf(ctx, expr)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).([]*model.UserGroupsEntity)
		}
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(context.Context, *model.UserGroupsFindExpr) error); ok {
		r1 = rf(ctx, expr)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// Insert provides a mock function with given fields: ctx, ent
func (_m *UserGroupsProvider) Insert(ctx context.Context, ent *model.UserGroupsEntity) (*model.UserGroupsEntity, error) {
	ret := _m.Called(ctx, ent)

	var r0 *model.UserGroupsEntity
	if rf, ok := ret.Get(0).(func(context.Context, *model.UserGroupsEntity) *model.UserGroupsEntity); ok {
		r0 = rf(ctx, ent)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(*model.UserGroupsEntity)
		}
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(context.Context, *model.UserGroupsEntity) error); ok {
		r1 = rf(ctx, ent)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// Set provides a mock function with given fields: ctx, userID, groupIDs
func (_m *UserGroupsProvider) Set(ctx context.Context, userID int64, groupIDs []int64) (int64, int64, error) {
	ret := _m.Called(ctx, userID, groupIDs)

	var r0 int64
	if rf, ok := ret.Get(0).(func(context.Context, int64, []int64) int64); ok {
		r0 = rf(ctx, userID, groupIDs)
	} else {
		r0 = ret.Get(0).(int64)
	}

	var r1 int64
	if rf, ok := ret.Get(1).(func(context.Context, int64, []int64) int64); ok {
		r1 = rf(ctx, userID, groupIDs)
	} else {
		r1 = ret.Get(1).(int64)
	}

	var r2 error
	if rf, ok := ret.Get(2).(func(context.Context, int64, []int64) error); ok {
		r2 = rf(ctx, userID, groupIDs)
	} else {
		r2 = ret.Error(2)
	}

	return r0, r1, r2
}
